### ISSUE TITLE
Questions about usage

### ISSUE THREAD

**Issue by dcosson** [bonovoxly/ansible_snippets#1]
Hey, just getting started with ansible on EC2 and looking at different ways to set things up and I came across your blog post and this repo. Curious about a few things:

- is there a way to set a different hostname variable than the `ansible_ssh_host`? With the add_host module I don't see a way to set a human readable name but still use the IP address for connecting, the way you can with ec2.py dynamic inventory.

- how would you run ad-hoc commands this way? i.e. use the `ansible` command instead of `ansible-playbook`

**Comment by bonovoxly** [comment]
that's exactly what ansible's `add_host` can do - http://docs.ansible.com/ansible/add_host_module.html

the task allows you to set a human readable value to an IP address.  let's say you wanted to refer to an IP address of `10.1.1.2` as `bob`:

```
- add_host:
    hostname: 10.1.1.2
    groups: bob
```

later in your playbook you can reference `bob`:

```
- hosts: bob
  become: yes
  gather_facts: yes

  roles:
    - role: your_role_or_your_tasks
```

I'm not sure how well it would work;  `add_host` only adds the group to memory, so it's only valid for that run/play.

**Comment by dcosson** [comment]
That makes sense for how the groups work. But the run output will show the ip address/hostnames instead of the human readable tag name, but that's not a big deal.

I think I've figured out a setup that I like a little better, which involves running ansible in both push and pull modes. I have a push playbook (which runs some minimal setup on hosts: all) and a pull playbook (which runs some minimal setup on hosts: localhost, it does an add_host to add its own node with some groups based on my conventions). Both of these playbooks do a top-level includes to include my shared playbook, which has all of the configuration of what roles to run for each group.

One useful trick that I've figured out with this - the different ansible modules including `ec2.py` inventory, `ec2_facts` and `ec2_remote_facts` all use slightly different names for some of their facts which makes them incompatible. So even though I'm using ec2.py inventory for my push-mode runs, I only use it for grouping & discovery but ignore all of the generated hostvars in favor of running `ec2_facts` on all hosts at the beginning of the playbook. In my pull mode playbook, I can also run `ec2_facts` on localhost. Then in my shared playbook, I'm working with the same set of fact names and everything is compatible.

Anyway, it was helpful reading your blog post & looking through this repo. Thanks!
